internal/teststep/config.go

```diff
@@ -21,7 +21,7 @@ const (
 )
 
 var (
-	providerConfigBlockRegex  = regexp.MustCompile(`provider "?[a-zA-Z0-9_-]+"? {`)
+	providerConfigBlockRegex  = regexp.MustCompile(`provider(\s*"[a-zA-Z0-9_-]+"\s*|\s+[a-zA-Z0-9_-]+\s*){`)
```

Comment:
Nit: Might be good to document (or consider unit testing the regular expression itself, although your other testing seems to cover it well already) the various cases this is intended to catch since its now more complicated:

```suggestion
	// Expected to match:
	//     provider "example" {
	//     provider "example"{
	//     provider example {
	//     provider example{
	//     provider"example"{
	providerConfigBlockRegex  = regexp.MustCompile(`provider(\s*"[a-zA-Z0-9_-]+"\s*|\s+[a-zA-Z0-9_-]+\s*){`)
```